Assets/Scripts/CharacterControllers/Melody/States/MovingState.cs

```diff
@@ -0,0 +1,73 @@
+﻿namespace Melody.States
+{
+    using UnityEngine;
+
+    public class MovingState : MelodyState
+    {
+        public MovingState(MelodyController controller) : base(controller) { }
+
+        protected override void Enter() { }
+
+        public override void OnUpdate(float time)
+        {
+            base.OnUpdate(time);
+
+            //Check For Attack
+            if (melodyController.input.AttackButtonDown())
+            {
+
+                ableToExit = true;
+                nextState = new AttackState(melodyController);
+            }
+            else if (melodyController.input.ParryButtonDown())
+            {
+                ableToExit = true;
+                nextState = new CounterState(melodyController);
+            }
+            else if (melodyController.input.DodgeButtonDown())
+            {
+                ableToExit = true;
+                nextState = new DodgeState(melodyController);
+            }
+            else if (melodyController.input.GetHorizontalMovement() == 0 && melodyController.input.GetVerticalMovement() == 0)
+            {
+                ableToExit = true;
+                nextState = new IdleState(melodyController);
+            }
+
+            melodyController.animator.SetFloat("Move", melodyController.Move.magnitude / 1);
+            melodyController.Move *= melodyController.MaxSpeed;
+            RotatePlayer(3);
+
+            melodyController.Move = melodyController.Move * time * 100;
```

Comment:
RE:

Should we perhaps store this movement multiplier somewhere in the MelodyController and grab it from there? I'd prefer to keep Melody's movement values all in one place to make it easy to tweak things.